src/test/ui/issues/issue-51719.stderr

```diff
@@ -1,8 +1,11 @@
 error[E0728]: `await` is only allowed inside `async` functions and blocks
   --> $DIR/issue-51719.rs:10:19
    |
+LL | async fn foo() {}
+   | -------------- this function is not `async`
```

Comment:
This doesn't look correct - shouldn't this point to `fn make_generator` instead?

Reply:
It should point at the span of the enclosing closure. (The `.stderr` is incorrect, yeah)

Reply:
That is:
```
L |     let _gen = || foo.await;
  |                   ^^^^^^^^^ only allowed inside `async` functions and blocks
  |                ------------ this closure is not `async`
```

Reply:
Fixed